packages/client/src/lib.js

```diff
@@ -208,12 +208,33 @@ class Web3Storage {
   /**
    * @param {Service} service
    * @param {CIDString} cid
-   * @returns {Promise<CIDString>}
+   * @returns {Promise<CIDString|undefined>}
    */
   /* c8 ignore next 4 */
   static async delete ({ endpoint, token }, cid) {
-    console.log('Not deleting', cid, endpoint, token)
-    throw Error('.delete not implemented yet')
+    const url = new URL(`/user/uploads/${cid}`, endpoint)
+    try {
+      const res = await fetch(url.toString(), {
+        method: 'DELETE',
+        headers: Web3Storage.headers(token)
+      })
+      if (res.status === 404) {
+        return undefined
+      }
+      if (!res.ok) {
+        throw new Error(res.statusText)
+      }
+      const response = await res.json()
+      if (response.response.errors) {
+        console.error(response.response.errors)
+        return undefined
+      }
+
+      return response
+    } catch (e) {
+      console.error(e)
+      return undefined
```

Comment:
humm, wondering if we should fail in case of no delete found. what do you think?

Reply:
It gracefully fails, as it says:
![image](https://user-images.githubusercontent.com/24696635/130647746-ae809949-8ddb-4f3c-ba2c-695849a0fc39.png)

Reply:
yes, if fails gracefully in the CLI, but I am concerned on the client UX. What if we go with an approach like JS built in Map [return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete#return_value)? We would return `true` if deleted, `false` otherwise. This would fit well with our `client.get` behaviour.

Reply:
Gotcha, changing it